doc/netplan.md

```diff
@@ -356,9 +440,17 @@ wpasupplicant installed if you let the ``networkd`` renderer handle wifi.
      supported properties:
 
      ``password`` (scalar)
-     :    Enable WPA2 authentication and set the passphrase for it. If not
-          given, the network is assumed to be open. Other authentication modes
-          are not currently supported.
+     :    Enable WPA2 authentication and set the passphrase for it. If neither
+          this nor an ``auth`` block are given, the network is assumed to be
+          open. The setting
+
+              password: "S3kr1t"
+
+          is equivalent to
+
+              auth:
+                key-mgmt: wpa-psk
+                psk: "S3kr1t"
```

Comment:
@cyphermox do you think we should be marking 'password' as deprecated in this context?

Reply:
IMHO password: shouldn't be deprecated, it's a very convenient shorthand for
"""
key-mgmt: wpa-psk
psk: "password"
"""

It certainly makes it a lot easier to remember how to quickly write up a netplan for the most common wireless (WPA personal) without having to resort to doc. At least, that's a clear improvement on my personal workflow for writing wpasupplicant config.